### PR TITLE
[SASS-3849] Refactor Controller for Page: 'Did you pay non-UK tax on the amount that resulted in a surcharge?"

### PR DESCRIPTION
**Pull request by erasmos-hmrc**
# Checklist

Jira subtask: SASS-3889

## Developer: Before Asking Others To Review Your Code

Have you ...

- [X]  Had a QA verify that all acceptance criteria have been met, with comment(s) in the ticket? 
- [X]  Hosted a group review of your implementation?
- [X]  Rebased against `main`? 
- [X]  Ensured that your commit(s) contain the ticket key?
- [X]  Ensured that your branch is named after the ticket key?
- [X]  Successfully run the unit tests?
- [X]  Successfully run the integration tests?
- [x]  Successfully run the journey tests? If not, please add a comment explaining why it wasn't necessary.
- [x]  Ensured that the PR job has passed.

## Reviewer: Before Reviewing

Have you ...

- [x]  Pulled the branch down?
- [x]  Assigned yourself to this PR?
- [x]  Moved the Jira *subtask* to “in review”?
- [x]  Ensured that it is rebased against `main`?

## Developer: After Approval

Have you ...

- [ ]  Rebased against `main`?
- [ ]  Moved the Jira *subtask* to “in pipeline”?

## Developer: After Merging

Have you ...

- [ ]  Deleted the branch?
- [ ]  Scheduled a demo? If not, please add a comment.

### CODE REVIEW

**Comment by erasmos-hmrc** [comment]
> Successfully run the journey tests? If not, please add a comment explaining why it wasn't necessary.

Note that I (well, Anthony) still needs to run these.

**Comment by erasmos-hmrc** [comment]
> Successfully run the journey tests? If not, please add a comment explaining why it wasn't necessary.

@walesoares ran these for me; although he had trouble with the CIS tests (as that service wouldn't start locally), all others had passed including those for pensions, of course.

File: app/common/MessageKeys.scala [hunk 1]
```diff
@@ -18,50 +18,88 @@ package common
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/common/MessageKeys.scala:18]: Good idea. Will do. Will help compact the file, as well.

File: it/controllers/pensions/unauthorisedPayments/NonUkTaxOnAmountNotSurchargeControllerISpec.scala [hunk 2]
```diff
@@ -81,7 +81,7 @@ class NonUkTaxOnAmountNotSurchargeControllerISpec extends YesNoAmountControllerS
                 radioButtonForYes = uncheckedExpectedRadioButton("Yes"),
```
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/controllers/pensions/unauthorisedPayments/NonUkTaxOnAmountNotSurchargeControllerISpec.scala:81]: Good catch - will change.

File: app/common/MessageKeys.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply app/common/MessageKeys.scala:18]: Changed.

File: it/controllers/pensions/unauthorisedPayments/NonUkTaxOnAmountNotSurchargeControllerISpec.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (erasmos-hmrc) [reply it/controllers/pensions/unauthorisedPayments/NonUkTaxOnAmountNotSurchargeControllerISpec.scala:81]: Changed.

**Review by salman-a-dev — APPROVED** [review]

**Comment by erasmos-hmrc** [comment]
> Scheduled a demo? If not, please add a comment.

I don't see a need for a demo as nothing has changed for the user.

**Pull request merged**